### PR TITLE
1001 link library

### PR DESCRIPTION
**Pull request by mdinsmore**
@pkimber @gbushell can you review the link library please - if running it, should work on example app - log in, browse to settings | link library - you may need to update your personal dev_  settings for the debug toolbar (see mine)

### CODE REVIEW

**Comment by pkimber** [comment]
Hi @mdinsmore 
Looks excellent...

1. Do we need link categories on the settings menu?  It makes sense to me to leave the button in the library.  I was confused because I clicked on link categories from the library, then back and ended up back at settings.
2. Do we need the URL in the link library?  I think it makes the view more complicated than it needs to be.  If we do need the URL, then perhaps make it small and further to the right?
3. Can we make the *target* into a link, so the user can download and view the document?

Very nice to see this.  It will be a great update to the `block` app.
:+1:

File: block/models.py [hunk 1]
```diff
@@ -1363,6 +1365,30 @@ def is_external(self):
         return bool(self.link_type == self.URL_EXTERNAL)
 
     @property
+    def blocks_used(self):
+        collector = NestedObjects(using='default')
```
Comment (pkimber) [inline block/models.py:1386]: @mdinsmore would you mind adding a short `docstring` to this describing what you are doing with `NestedObjects` please?  I can't see Django documentation for the class.
Thank you

**Comment by mdinsmore** [comment]
Hi Patrick,

I agree on the image and link categories - we don't need them on the menu -
that simplifies the menu. - Done

The idea of the on the Url: column is this is an address for this link that
will not change if, for example, the file is updated or the link is changed
to point to another page.   Without this if the document is changed any
link you put in text would need to be updated.

I've added a docstring for the to the blocks_used.

Have a good

**Comment by mdinsmore** [comment]
oops that was supposed to say "Have a good weekend"

**Comment by pkimber** [comment]
Thank you @mdinsmore 
I think I am too tired to think properly.  Can we review together on Monday please?
Hope you have a great weekend too...

**Review by pkimber — APPROVED** [review]

**Pull request merged**